src/components/IKImage/IKImage.js

```diff
@@ -33,35 +33,88 @@ class IKImage extends ImageKitComponent {
     return state;
   }
 
-  lqipload(quality) {
+  lqipload(quality, blur, threshold) {
+    const finalQuality = parseInt((quality || threshold), 10) || 20;
+    const finalBlur = parseInt(blur, 10) || 6;
     let url = this.state.url;
     let lqip = "";
     if (this.props.path !== undefined) {
       let newUrl = url.split("tr:");
       if (newUrl[0] === url) {
         let newUrl = url.split("/");
-        newUrl = `${newUrl[0]}//${newUrl[2]}/${newUrl[3]}/tr:q-${quality}/${newUrl[4]}`;
+        newUrl = `${newUrl[0]}//${newUrl[2]}/${newUrl[3]}/tr:q-${finalQuality},bl-${finalBlur}/${newUrl[4]}`;
         lqip = `${newUrl}`;
       } else {
-        newUrl = `${newUrl[0]}tr:q-${quality},${newUrl[1]}`;
+        newUrl = `${newUrl[0]}tr:q-${finalQuality},bl-${finalBlur},${newUrl[1]}`;
         lqip = `${newUrl}`;
       }
     } else {
       if(url.includes("tr")){
-        lqip = `${url}&q-${quality}`;
+        lqip = `${url}&q-${finalQuality},bl-${finalBlur}`;
       }
       else {
-        lqip = `${url}&tr=q-${quality}`;
+        lqip = `${url}&tr=q-${finalQuality},bl-${finalBlur}`;
       }
 
     }
     return lqip;
   }
 
+  getFinalImageSrcURL(loading, lqip){
+
+    /*
+    No lazy loading no lqip
+      src=originalImage
+    No lazy loading lqip
+      src=lqip
+      src=originalImage (when loaded)
+    lazy loading and no lqip
+      src=''
+      onIntersect:
+      src=originalImage
+    lazy loading and lqip
+      src=lqip
+      onIntersect:
+      src=originalImage (when loaded)
+   */
+
+    const {
+      imageLoaded,
+      isIntersecting,
+    } = this.state;
+
+    const {
+      url
+    } = this.state;
+
+    if (loading !== "lazy" && lqip === null) {
+      return url;
+    } else if (loading !== "lazy" && lqip && lqip.active) {
+      if (imageLoaded) {
+        return url;
+      } else {
+        return lqipload(lqip.quality, lqip.blur, lqip.threshold);
+      }
+    } else if (loading === "lazy" && lqip === null) {
+      if (isIntersecting) {
+        return url;
+      } else {
+        return "";
+      }
+    } else if (loading === "lazy" && lqip && lqip.active) {
+      if (isIntersecting && imageLoaded) {
+        return url;
+      } else {
+        return lqipload(lqip.quality, lqip.blur, lqip.threshold);
+      }
+    }
+  }
+
   componentDidMount() {
     const imageObserver = new IntersectionObserver(function (entry, observer) {
       if (entry[0].isIntersecting) {
         let image = entry[0].target;
+        this.setState({isIntersecting: true});
         image.src = url;
```

Comment:
Why are we setting `image.src` to url here?

We are relying on the onLoad event but we don't know if it will be triggered after low quality image is loaded or the original src. We can make this logic explicit as in Vue.js by triggering the triggerOriginalImageLoad function appropriately.